scripts/remove_stationary_segments.py

```
@@ -0,0 +1,259 @@
+
+try:
+    from geopy.distance import geodesic
+    GEOPY_AVAILABLE = True
+except ImportError:
+    GEOPY_AVAILABLE = False
+    print("⚠️  geopy not installed. Install with: pip install geopy")
+
+
+def parse_gpx_track(gpx_file: str) -> List[dict]:
+    """Parse GPX file and extract trackpoints with timestamps."""
+    tree = ET.parse(gpx_file)
```

Ruff (0.15.14)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)